### ISSUE TITLE
BasicObject.singleton_class behaviour different from MRI

### ISSUE THREAD

**Issue by dkubb** [jruby/jruby#1043]
In MRI the following is `true`, but it is `false` under JRuby:

``` ruby
Numeric.kind_of?(BasicObject.singleton_class)
```

I originally came across this difference when doing the following refactoring:

``` diff
-        object.respond_to?(:ancestors) &&
-        object.ancestors.include?(primitive)
+        object.kind_of?(primitive.singleton_class)
```

**Comment by MSNexploder** [comment]
This is fixed in JRuby 9000.pre2 but still present in 1.7.20.

**Comment by enebo** [comment]
This works in 9k and we will never fix this in 1.7.x as it is winding down....resolving.
